Chessington.GameEngine/Pieces/Knight.cs

```diff
@@ -10,7 +10,20 @@ public Knight(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var moves = new List<Square>();
+            var currentSquare = board.FindPiece(this);
+            var moveX = 2;
+            var moveY = 1;
+            moves.AddIfOnBoard(new Square(currentSquare.Row + moveX, currentSquare.Col + moveY));
+            moves.AddIfOnBoard(new Square(currentSquare.Row + moveX, currentSquare.Col - moveY));
+            moves.AddIfOnBoard(new Square(currentSquare.Row - moveX, currentSquare.Col + moveY));
+            moves.AddIfOnBoard(new Square(currentSquare.Row - moveX, currentSquare.Col - moveY));
+            moves.AddIfOnBoard(new Square(currentSquare.Row + moveY, currentSquare.Col + moveX));
```

Comment:
The `moveX` and `moveY` variable names don't really work for these last four statements unfortunately 🙂  I'd probably either name them something like `twoSteps` and `oneStep`, or just use 2 and 1 without assigning them to variables 🙂

Reply:
Yeah I was wondering with that. The main reason was for if we ever wanted to change the knight rules it would be easier to change a variable than all the statements. But since we're just coding basic chess that was probably unnecessary

Reply:
Good thinking. However I think if you're changing the knight rules then you'd be likely to need to change more than just those constants. And yes as you've pointed out, since we're implementing some rules that haven't changed in a long time I think it's safe to assume they're not going to change 🙂